kebechet_administrator.py

```diff
@@ -0,0 +1,159 @@
+        package_name=Configuration.PACKAGE_NAME,
+        index_url=Configuration.PACKAGE_INDEX,
+        os_name=os_name,
+        os_version=os_version,
+        python_version=python_version,
+    )  # We query without the package_version.
+    for key in repositories.keys():
+        repo_info = repositories[key]
+        # Construct the message input
+        if repo_info.get("private"):
+            continue  # We skip for private repo's.
+        if semver.compare(repo_info.get("package_version"), Configuration.PACKAGE_VERSION):
+            continue  # We dont schedule, if the package version > version of the solved package version.
+        message_input = {
+            "component_name": {"type": "str", "value": __COMPONENT_NAME__},
+            "service_version": {"type": "str", "value": __service_version__},
+            "url": {"type": "str", "value": _URL_PREFIX + key},
+            "service_name": {"type": "str", "value": "github"},
+            "installation_id": {"type": "str", "value": repo_info.get("installation_id")},
+        }
+
+        # We store the message to put in the output file here.
+        output_messages.append({"topic_name": "thoth.kebechet-run-url-trigger", "message_contents": message_input})
+
+
+def _handle_package_issue(Configuration):  # noqa: N803
+    """Handle all the messages for which Kebechet needs to run on all repos associated."""
+    # We getch all the Kebechet repos using a non optimal package(missing package or CVE or missing version)
+    repositories: Dict[str, Dict] = GRAPH.get_kebechet_github_installations_info_for_python_package_version(
+        package_name=Configuration.PACKAGE_NAME,
+        version=Configuration.PACKAGE_VERSION,
+        index_url=Configuration.PACKAGE_INDEX,
+    )
+    # The keys represent the repo names and the value dictionary is the details for the repo.
+    for key in repositories.keys():
+        repo_info = repositories[key]
+        # Construct the message input
+        if repo_info.get("private"):
+            continue  # We skip for private repo's.
+        message_input = {
+            "component_name": {"type": "str", "value": __COMPONENT_NAME__},
+            "service_version": {"type": "str", "value": __service_version__},
+            "url": {"type": "str", "value": _URL_PREFIX + key},
+            "service_name": {"type": "str", "value": "github"},
+            "installation_id": {"type": "str", "value": repo_info.get("installation_id")},
+        }
+
+        # We store the message to put in the output file here.
+        output_messages.append({"topic_name": "thoth.kebechet-run-url-trigger", "message_contents": message_input})
+
+
+# This handler dispatches the specific method based on a paticular message.
+_message_handler = {
+    "SolvedPackageMessage": _handle_solved_message,
+    "HashMismatchMessage": _handle_package_issue,
+    # "MissingPackageMessage": _handle_package_issue, # To be implemented.
```

Comment:
this is implemented actually: https://github.com/thoth-station/messaging/blob/master/thoth/messaging/missing_package.py